### PR TITLE
Add field names to pallet Event variants

### PR DESCRIPTION
**Pull request by Wizdave97**
## Description
This PR changes the pallet events variants from tuples to structs with named fields

Part implementation for  #9903

## Pallets updated
Sorted alphabetically in the frame folder  assets - proxy

polkadot companion: paritytech/polkadot#4262
cumulus companion: paritytech/cumulus#749

Polkadot address: 1xazYr2XULzE3kjWCbuzKNcykfbjdgJGeSD6n3RvhdSikJf

### CODE REVIEW

**Review by kianenigma — DISMISSED** [review]
Did you run `cargo +nightly fmt`?

File: frame/assets/src/benchmarking.rs [hunk 1]
```diff
@@ -155,15 +155,15 @@ benchmarks_instance_pallet! {
 		T::Currency::make_free_balance_be(&caller, DepositBalanceOf::<T, I>::max_value());
 	}: _(SystemOrigin::Signed(caller.clone()), Default::default(), caller_lookup, 1u32.into())
 	verify {
-		assert_last_event::<T, I>(Event::Created(Default::default(), caller.clone(), caller).into());
+		assert_last_event::<T, I>(Event::Created{asset_id: Default::default(), creator: caller.clone(), owner: caller}.into());
```
[the remark this answers is not in the document]
Comment (kianenigma) [reply frame/assets/src/benchmarking.rs:158]: rest of this file needs manual fix.

**Review by kianenigma — COMMENTED** [review]
LGTM, this will probably be breaking change for Jaco and Polkadot UI?

File: frame/assets/src/benchmarking.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wizdave97) [reply frame/assets/src/benchmarking.rs:158]: I'll work on that later today

**Comment by athei** [comment]
> LGTM, this will probably be breaking change for Jaco and Polkadot UI?

You still are blocking this PR with a "changes requested" PR.

**Comment by Wizdave97** [comment]
> > LGTM, this will probably be breaking change for Jaco and Polkadot UI?
> 
> You still are blocking this PR with a "changes requested" review, though.

Where do I need to make changes for polkadot UI?

**Comment by athei** [comment]
The failing test is concerning. As if there is an error in here somehow. Can't find it. Did you check out the failing test?

**Comment by Wizdave97** [comment]
> The failing test is concerning. As if there is an error in here somehow. Can't find it. Did you inspect the failing test?

Checking it out now

**Comment by Wizdave97** [comment]
> The failing test is concerning. As if there is an error in here somehow. Can't find it. Did you inspect the failing test?

Fixed it up

File: frame/balances/src/tests_reentrancy.rs [hunk 2]
```diff
@@ -197,15 +204,22 @@ fn transfer_dust_removal_tst2_should_work() {
...
+			amount: 50,
+		}));
+		System::assert_has_event(Event::Balances(crate::Event::Deposit { who: 1, amount: 50 }));
 	});
 }
 
 #[test]
 fn repatriating_reserved_balance_dust_removal_should_work() {
-	ExtBuilder::default().existential_deposit(100).build().execute_with(|| {
+	ExtBuilder::default().existential_deposit(200).build().execute_with(|| {
```
Comment (athei) [inline frame/balances/src/tests_reentrancy.rs:222]: This is concerning. Behavior should not change with this PR. Please revert this back and try if it works.

Comment (Wizdave97) [reply frame/balances/src/tests_reentrancy.rs:222]: Yeah, that was an oversight, the main error was from the assertion, I’ll revert this.

File: frame/staking/src/slashing.rs [hunk 3]
```diff
@@ -190,7 +190,7 @@ pub(crate) struct SpanRecord<Balance> {
 impl<Balance> SpanRecord<Balance> {
 	/// The value of stash balance slashed in this span.
 	#[cfg(test)]
-	pub(crate) fn amount_slashed(&self) -> &Balance {
+	pub(crate) fn amount(&self) -> &Balance {
```
Comment (athei) [inline frame/staking/src/slashing.rs:193]: Unrelated but good call.

**Review by athei — APPROVED** [review]
LGTM. Thanks for doing this.

**Comment by KiChjang** [comment]
It looks like this PR needs a polkadot companion and a cumulus companion as well, we can't merge this until the Event variants are also updated on those two repositories.

**Comment by Wizdave97** [comment]
> It looks like this PR needs a polkadot companion and a cumulus companion as well, we can't merge this until the Event variants are also updated on those two repositories.

Okay, I'll create PRs for those also

File: frame/balances/src/tests.rs [hunk 4]
```diff
@@ -342,7 +342,7 @@ macro_rules! decl_tests {
 		fn balance_works() {
 			<$ext_builder>::default().build().execute_with(|| {
 				let _ = Balances::deposit_creating(&1, 42);
-				System::assert_has_event(Event::Balances(crate::Event::Deposit(1, 42)));
+				System::assert_has_event(Event::Balances(crate::Event::Deposit{who: 1, amount: 42}));
```
Comment (emostov) [inline frame/balances/src/tests.rs:345]:
```suggestion
				System::assert_has_event(Event::Balances(crate::Event::Deposit{ who: 1, amount: 42 }));
```
Same change as requested in https://github.com/paritytech/substrate/pull/9993/files#r739342924 . Goes for everything in this macro

File: frame/bounties/src/benchmarking.rs [hunk 5]
```diff
@@ -172,7 +172,7 @@ benchmarks! {
 		let bounty_id = BountyCount::<T>::get() - 1;
 	}: close_bounty(RawOrigin::Root, bounty_id)
 	verify {
-		assert_last_event::<T>(Event::BountyCanceled(bounty_id).into())
+		assert_last_event::<T>(Event::BountyCanceled{index: bounty_id}.into())
```
Comment (emostov) [inline frame/bounties/src/benchmarking.rs:175]:
```suggestion
		assert_last_event::<T>(Event::BountyCanceled{ index: bounty_id }.into())
```

Comment (emostov) [reply frame/bounties/src/benchmarking.rs:175]: Same in the rest of the file

File: frame/collective/src/benchmarking.rs [hunk 6]
```diff
@@ -128,7 +128,7 @@ benchmarks_instance_pallet! {
 		let proposal_hash = T::Hashing::hash_of(&proposal);
 		// Note that execution fails due to mis-matched origin
 		assert_last_event::<T, I>(
-			Event::MemberExecuted(proposal_hash, Err(DispatchError::BadOrigin)).into()
+			Event::MemberExecuted{proposal_hash, result: Err(DispatchError::BadOrigin)}.into()
```
Comment (emostov) [inline frame/collective/src/benchmarking.rs:131]:
```suggestion
			Event::MemberExecuted{ proposal_hash, result: Err(DispatchError::BadOrigin) }.into()
```

Comment (emostov) [reply frame/collective/src/benchmarking.rs:131]: Same for the rest of the file

File: frame/democracy/src/benchmarking.rs [hunk 7]
```diff
@@ -774,7 +774,7 @@ benchmarks! {
 	}: enact_proposal(RawOrigin::Root, proposal_hash, 0)
 	verify {
 		// Fails due to mismatched origin
-		assert_last_event::<T>(Event::<T>::Executed(0, Err(BadOrigin.into())).into());
+		assert_last_event::<T>(Event::<T>::Executed{ref_index: 0, result: Err(BadOrigin.into())}.into());
```
Comment (emostov) [inline frame/democracy/src/benchmarking.rs:777]:
```suggestion
		assert_last_event::<T>(Event::<T>::Executed{ ref_index: 0, result: Err(BadOrigin.into()) }.into());
```

File: frame/election-provider-multi-phase/src/lib.rs [hunk 8]
```diff
@@ -1026,18 +1029,18 @@ pub mod pallet {
 		/// solution is unsigned, this means that it has also been processed.
 		///
 		/// The `bool` is `true` when a previous solution was ejected to make room for this one.
```
Comment (emostov) [inline frame/election-provider-multi-phase/src/lib.rs:1031]:
```suggestion
		/// The `prev_ejected` is `true` when a previous solution was ejected to make room for this one.
```

File: frame/identity/src/benchmarking.rs [hunk 9]
```diff
@@ -153,7 +153,7 @@ benchmarks! {
 		};
 	}: _(RawOrigin::Signed(caller.clone()), Box::new(create_identity_info::<T>(x)))
 	verify {
-		assert_last_event::<T>(Event::<T>::IdentitySet(caller).into());
+		assert_last_event::<T>(Event::<T>::IdentitySet{who: caller}.into());
```
Comment (emostov) [inline frame/identity/src/benchmarking.rs:156]:
```suggestion
		assert_last_event::<T>(Event::<T>::IdentitySet{ who: caller }.into());
```

Same for rest of file

File: frame/offences/benchmarking/src/lib.rs [hunk 10]
```diff
@@ -315,13 +315,13 @@ benchmarks! {
 			<T as StakingConfig>::Event::from(StakingEvent::<T>::Slashed(id, BalanceOf::<T>::from(slash_amount)))
 		);
 		let balance_slash = |id| core::iter::once(
-			<T as BalancesConfig>::Event::from(pallet_balances::Event::<T>::Slashed(id, slash_amount.into()))
+			<T as BalancesConfig>::Event::from(pallet_balances::Event::<T>::Slashed{who: id, amount: slash_amount.into()})
```
Comment (emostov) [inline frame/offences/benchmarking/src/lib.rs:318]:
```suggestion
			<T as BalancesConfig>::Event::from(pallet_balances::Event::<T>::Slashed{ who: id, amount: slash_amount.into() })
```

Same for the rest of the file

File: frame/proxy/src/benchmarking.rs [hunk 11]
```diff
@@ -86,7 +86,7 @@ benchmarks! {
 		let call: <T as Config>::Call = frame_system::Call::<T>::remark { remark: vec![] }.into();
 	}: _(RawOrigin::Signed(caller), real, Some(T::ProxyType::default()), Box::new(call))
 	verify {
-		assert_last_event::<T>(Event::ProxyExecuted(Ok(())).into())
+		assert_last_event::<T>(Event::ProxyExecuted{result: Ok(())}.into())
```
Comment (emostov) [inline frame/proxy/src/benchmarking.rs:89]:
```suggestion
		assert_last_event::<T>(Event::ProxyExecuted{ result: Ok(()) }.into())
```
Same for rest of file

**Review by emostov — COMMENTED** [review]
Looks good overall - just incorporate the formatting changes within the macros and it should be good to go

**Comment by Wizdave97** [comment]
Polkadot Companion PR https://github.com/paritytech/polkadot/pull/4262
Work on cumulus companion still in progress

**Comment by KiChjang** [comment]
@Wizdave97 I've added them to your PR description. The CI bots will only read that to determine where the companion PRs are, instead of new github comments.

**Comment by Wizdave97** [comment]
> @Wizdave97 I've added them to your PR description. The CI bots will only read that to determine where the companion PRs are, instead of new github comments.

@KiChjang  Do i need to rerun the CI jobs again after this?

**Comment by KiChjang** [comment]
@Wizdave97 Shouldn't need to; if I'm not mistaken, there is a github hook on every edit of the PR description box, and the CI bots pick up on that.

**Comment by Wizdave97** [comment]
> @Wizdave97 Shouldn't need to; if I'm not mistaken, there is a github hook on every edit of the PR description box, and the CI bots pick up on that.

There's a failing test, but it's not due to the changes from this PR

**Comment by Wizdave97** [comment]
> @Wizdave97 Shouldn't need to; if I'm not mistaken, there is a github hook on every edit of the PR description box, and the CI bots pick up on that.

@KiChjang  All done

**Comment by Wizdave97** [comment]
@shawntabrizi  @kianenigma  Could you please review  this and it's companon PRs

**Review by thiolliere — APPROVED** [review]
looks good to me

**Comment by thiolliere** [comment]
bot merge

**Comment by Wizdave97** [comment]
> /tip large

Thank you.

**Pull request merged**